x/distribution/keeper/custom_msg_server_test.go

```diff
@@ -0,0 +1,76 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/initia-labs/initia/x/distribution/keeper"
+	customTypes "github.com/initia-labs/initia/x/distribution/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDepositValidatorRewardsPool(t *testing.T) {
+	ctx, keepers := createDefaultTestInput(t)
+	customMsgServer := keeper.NewCustomMsgServerImpl(keepers.DistKeeper)
+	_, _, depositor := keyPubAddr()
+	keepers.Faucet.Mint(ctx, depositor, types.NewCoin("uinit", math.NewInt(10)))
+	val := createValidatorWithBalance(ctx, keepers, 100, 100, 1)
+
+	cases := []struct {
+		name   string
+		msg    *customTypes.MsgDepositValidatorRewardsPool
+		resp   *customTypes.MsgDepositValidatorRewardsPoolResponse
+		errMsg string
+	}{
+		{
+			name: "success",
+			msg: &customTypes.MsgDepositValidatorRewardsPool{
+				Depositor:        depositor.String(),
+				ValidatorAddress: val.String(),
+				Denom:            "uinit",
+				Amount:           types.NewCoins(types.NewCoin("uinit", math.NewInt(10))),
+			},
+
+			resp:   &customTypes.MsgDepositValidatorRewardsPoolResponse{},
+			errMsg: "",
+		},
+		{
+			name: "invalid depositor address",
+			msg: &customTypes.MsgDepositValidatorRewardsPool{
+				Depositor:        "invalid",
+				ValidatorAddress: val.String(),
+				Denom:            "uinit",
+				Amount:           types.NewCoins(types.NewCoin("uinit", math.NewInt(10))),
+			},
+			resp:   &customTypes.MsgDepositValidatorRewardsPoolResponse{},
+			errMsg: "decoding bech32 failed: invalid bech32 string length 7",
+		},
+		{
+			name: "invalid validaor address",
+			msg: &customTypes.MsgDepositValidatorRewardsPool{
+				Depositor:        depositor.String(),
+				ValidatorAddress: "invalid",
+				Denom:            "uinit",
+				Amount:           types.NewCoins(types.NewCoin("uinit", math.NewInt(10))),
+			},
+			resp:   &customTypes.MsgDepositValidatorRewardsPoolResponse{},
+			errMsg: "VM aborted: location=0000000000000000000000000000000000000000000000000000000000000001::fungible_asset, code=65540",
```

Comment:
seems this error message coming from not enough balance. 

the test case 1 spends 10uinit, due to this the test case 3 failed with insufficient balance error.

Reply:
mint 20uinit